saul-core/src/main/java/edu/illinois/cs/cogcomp/saul/learn/SaulWekaWrapper.java

```diff
@@ -0,0 +1,587 @@
+                System.out.println("WekaWrapper: baseClassifier was null.");
+            e.printStackTrace();
+            System.exit(1);
+        }
+
+        trained = true;
+        wekaInstances = new Instances(name, attributeInfo, 0);
+        wekaInstances.setClassIndex(0);
+    }
+
+
+    /**
+     * Writes the settings of the classifier in use, and a string describing the classifier, if
+     * available.
+     **/
+    public void write(PrintStream out) {
+        out.print(name + ": ");
+        String[] options = baseClassifier.getOptions();
+        for (int i = 0; i < options.length; ++i)
+            out.println(options[i]);
+        out.println(baseClassifier);
+    }
+
+
+    /**
+     * Writes the learned function's internal representation in binary form.
+     *
+     * @param out The output stream.
+     **/
+    public void write(ExceptionlessOutputStream out) {
+        super.write(out);
+        out.writeBoolean(trained);
+
+        if (allowableValues == null)
+            out.writeInt(0);
+        else {
+            out.writeInt(allowableValues.length);
+            for (int i = 0; i < allowableValues.length; ++i)
+                out.writeString(allowableValues[i]);
+        }
+
+        ObjectOutputStream oos = null;
+        try {
+            oos = new ObjectOutputStream(out);
+        } catch (Exception e) {
+            System.err.println("Can't create object stream for '" + name + "': " + e);
+            System.exit(1);
+        }
+
+        try {
+            oos.writeObject(baseClassifier);
+            oos.writeObject(freshClassifier);
+            oos.writeObject(attributeInfo);
+            oos.writeObject(wekaInstances);
+        } catch (Exception e) {
+            System.err.println("Can't write to object stream for '" + name + "': " + e);
+            System.exit(1);
+        }
+    }
+
+
+    /**
+     * Reads the binary representation of a learner with this object's run-time type, overwriting
+     * any and all learned or manually specified parameters as well as the label lexicon but without
+     * modifying the feature lexicon.
+     *
+     * @param in The input stream.
+     **/
+    public void read(ExceptionlessInputStream in) {
+        super.read(in);
+        trained = in.readBoolean();
+        allowableValues = new String[in.readInt()];
+        for (int i = 0; i < allowableValues.length; ++i)
+            allowableValues[i] = in.readString();
+
+        ObjectInputStream ois = null;
+        try {
+            ois = new ObjectInputStream(in);
+        } catch (Exception e) {
+            System.err.println("Can't create object stream for '" + name + "': " + e);
+            System.exit(1);
+        }
+
+        try {
+            baseClassifier = (weka.classifiers.Classifier) ois.readObject();
+            freshClassifier = (weka.classifiers.Classifier) ois.readObject();
+            attributeInfo = (FastVector) ois.readObject();
+            wekaInstances = (Instances) ois.readObject();
```

Comment:
Curious if we need all the `wekaInstances` for just classifying after loading baseClassifier/freshClassifier. I didn't check if is being used in the classify path. Would be good to avoid writing/reading wekaInstances. [worried about model size].